lib/cn/cn_tree.c

```diff
@@ -1522,50 +1524,69 @@ cn_comp_update_split(
     }
 }
 
+static void
+cn_comp_kvset_append(
+    struct cn_compaction_work *work,
+    struct cn_tree_node       *node,
+    struct kvset              *kvset)
+{
+    struct cn_tree *tree = work->cw_tree;
+
+    if (work->cw_err)
+        return;
+
+    assert(node);
+    assert(kvset);
+
+    rmlock_wlock(&tree->ct_lock);
+
+    assert(node);
+    kvset_list_add(kvset, &node->tn_kvset_list);
+    node->tn_cgen++;
+
+    cn_tree_samp(tree, &work->cw_samp_pre);
+    cn_tree_samp_update_ingest(tree, node);
+    cn_tree_samp(tree, &work->cw_samp_post);
+
+    rmlock_wunlock(&tree->ct_lock);
+}
+
 /**
  * cn_comp_commit() - commit compaction operation to cndb log
  * See section comment for more info.
  */
 static void
 cn_comp_commit(struct cn_compaction_work *w)
 {
-    struct kvset **kvsets = 0;
+    struct kvset  **kvsets = 0;
     struct mbset ***vecs = 0;
-    uint *cnts = 0;
-    uint i, alloc_len;
-    const bool spill = (w->cw_action == CN_ACTION_SPILL);
-    const bool split = (w->cw_action == CN_ACTION_SPLIT);
-    const bool kcompact = (w->cw_action == CN_ACTION_COMPACT_K);
-    const bool use_mbsets = kcompact;
-    bool skip_commit = false, txn_nak = false;
-    void **cookiev = 0;
+    uint           *cnts = 0;
+    void          **cookiev = 0;
+    uint            alloc_len;
+    const bool      is_kcompact = (w->cw_action == CN_ACTION_COMPACT_K);
+    const bool      is_split = (w->cw_action == CN_ACTION_SPLIT);
+    const bool      is_spill = (w->cw_action == CN_ACTION_SPILL);
+    bool            use_mbsets = is_kcompact;
+    bool            txn_nak = false;
+    merr_t          err = 0;
+    uint            i, num_out = is_spill ? 1 : w->cw_outc;
```

Comment:
I think we gave up on this method of aligning local vars...

Reply:
Yeah, it was this way when I started working, but after rebasing to master it appeared as a change...I'll change this to how it is in master.